lightning-background-processor/src/fwd_batch.rs

```diff
@@ -0,0 +1,486 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+use core::sync::atomic::{AtomicU16, Ordering};
+use core::time::Duration;
+
+pub(crate) struct BatchDelay {
+	next_batch_delay_millis: AtomicU16,
+}
+
+impl BatchDelay {
+	pub(crate) fn new() -> Self {
+		let next_batch_delay_millis = AtomicU16::new(rand_batch_delay_millis());
+		Self { next_batch_delay_millis }
+	}
+
+	pub(crate) fn get(&self) -> Duration {
+		Duration::from_millis(self.next_batch_delay_millis.load(Ordering::Acquire) as u64)
```

Comment:
Where is BatchDelay used that requires this to be thread-safe? I looked at the bg processor loop and it all seems to be in sequence?